### ISSUE TITLE
wrong implementation of CW attack

### ISSUE THREAD

**Issue by CNOCycle** [MadryLab/mnist_challenge#10]
Hi authors,

cw attack in this repo is:
https://github.com/MadryLab/mnist_challenge/blob/1c61741549a423df33f732458012b4ee7b07e853/pgd_attack.py#L34

However, the original cw attack implementation is:
```
wrong_logit = tf.reduce_max((1-label_mask) * model.pre_softmax - 1e4*label_mask, axis=1)
```
I also found that in repo [MadryLab/cifar10_challenge](https://github.com/MadryLab/cifar10_challenge/) have fixed this issue by commit `958b8f`

**Comment by CNOCycle** [comment]
Another issue is that CW attacks on the leaderboard are based on "original" loss or loss in this repo?

**Comment by CNOCycle** [comment]
3rd issue is MadryLab/cifar10_challenge/issues/4 that clipping initial random perturbation.

But this doesn't fix in this repo.

**Comment by dtsip** [comment]
Thank you for pointing out these issues. Just fixed the definition of the CW loss and added initial clipping in https://github.com/MadryLab/mnist_challenge/commit/928fff56abe3bfb4d236976f6b1d7db8dce2f663 .

The change in CW loss should not change the performance of the models since this change is only relevant in cases where all logits are negative (which is quite unlikely typically).
